### DISCUSSION TITLE
Can't use any values other than 'button' for 'as' props

### DISCUSSION THREAD

**Discussion by itsfaqih** [ariakit/ariakit#824] [Questions] [answered]
Hi, I'm trying to build a dropdown component built on top of [Menu](https://reakit.io/docs/menu/), but there's a problem when I try to use the 'as' props, I couldn't use any value other than 'button':

![image](https://user-images.githubusercontent.com/48067039/104694086-41977a80-573d-11eb-9e80-a2db1de8abf0.png)

Here's how I made the dropdown item component:
```js
export function DropdownItem({
  children,
  className,
  ...props
}: ComponentProps<typeof MenuItem>) {
  const state = useContext(DropdownContext);
  return (
    <MenuItem className={classNames('w-full', className)} {...state} {...props}>
      {children}
    </MenuItem>
  );
}
```

**Comment by diegohaz** [comment, marked as answer]
@itsfaqih Could you make a TypeScript CodeSandbox of that so we can try it easier? But I think you should use `MenuProps` or something.

**Reply by itsfaqih** [reply]
Apparently it works fine, but it's still showing the errors in the code editor.

https://codesandbox.io/s/custom-dropdown-reakit-ioh4r

**Reply by diegohaz** [reply]
In that case, I think you'll have to type your `as` prop by yourself (example: https://codesandbox.io/s/typescript-as-prop-dicj8?file=/src/App.tsx) and combine it with either `Partial<MenuProps>` or `MenuHTMLProps`, `MenuItemProps` etc. (they're exported by reakit).

**Reply by itsfaqih** [reply]
sorry for the late reply. thank you very much! it works like a charm
